decomposer/multiagent_reasoner.py

```diff
@@ -26,431 +25,440 @@
 import time
 from pathlib import Path
 
-from decomposer.neuro_san_solver import NeuroSanSolver
+from neuro_san.interfaces.agent_session import AgentSession
 
-# Tuning knobs with environment variable overrides
-MAX_DEPTH: int = int(os.getenv("MAX_DEPTH", "5"))
-WINNING_VOTE_COUNT: int = int(os.getenv("WINNING_VOTE_COUNT", "2"))
-CANDIDATE_COUNT: int = (2 * WINNING_VOTE_COUNT) - 1
-NUMBER_OF_VOTES: int = (2 * WINNING_VOTE_COUNT) - 1
-SOLUTION_CANDIDATE_COUNT: int = (2 * WINNING_VOTE_COUNT) - 1
+from decomposer.neuro_san_agent_caller import NeuroSanAgentCaller
+from decomposer.session_manager import SessionManager
+from decomposer.trace_data import TraceData
 
-LOG_FAILURES_JSONL: str = os.getenv("LOG_FAILURES_JSONL")
-
-# Has 3 loosey-goosey fields: tree, decomposition, solve
-_trace_data: Any = threading.local()
-
-
-def setup() -> str:
-    log_level: str = os.getenv("LOG_LEVEL", "INFO").upper()
-
-    logging.basicConfig(level=log_level, format="[%(levelname)s] %(message)s", stream=sys.stderr)
-    logger = logging.getLogger()
-
-    log_file: str = None
-    log_dir: str = os.getenv("LOG_DIR")
-    if log_dir:
-        Path(log_dir).mkdir(parents=True, exist_ok=True)
-        log_file: str = Path(log_dir) / f"mr_{os.getpid()}_{threading.get_ident()}_{int(time.time())}.log"
-        file_handler = logging.FileHandler(log_file)
-        file_handler.setLevel(log_level)
-        file_handler.setFormatter(logging.Formatter("[%(levelname)s] %(message)s"))
-        logger.addHandler(file_handler)
-        logging.info(f"Logging to {log_file}")
-
-    os.environ["AGENT_MANIFEST_FILE"] = "./registries/manifest.hocon"
-    os.environ["AGENT_TOOL_PATH"] = "coded_tools"
-
-    return log_file
 
+class MultiAgentReasoner:
+    """
+    This guy is the main entry point for the multi-agent reasoner.
+    """
```

Comment:
Make mutliagent_reaonser.py global content into a class.
This is best viewed with "Hide whitespace".